backend/src/main/kotlin/app/ehrenamtskarte/backend/verification/service/CardVerifier.kt

```diff
@@ -15,19 +15,28 @@ val TIME_STEP: Duration = Duration.ofSeconds(30)
 const val TOTP_LENGTH = 6
 
 object CardVerifier {
-    fun verifyCardHash(project: String, cardHash: ByteArray, totp: Int, timezone: ZoneId): Boolean {
+    fun verifyCardHash(project: String, cardHash: ByteArray, totp: Int?, timezone: ZoneId): Boolean {
         val card = transaction { CardRepository.findByHashModel(project, cardHash) } ?: return false
-        return verifyCard(card, totp, timezone)
+        return if (totp != null) verifyCard(card, totp, timezone) else verifyStaticCard(card, timezone)
+    }
+
+    private fun verifyStaticCard(card: CardEntity, timezone: ZoneId): Boolean {
+        return !isExpired(card.expirationDay, timezone) &&
+            !card.revoked
     }
 
     private fun verifyCard(card: CardEntity, totp: Int, timezone: ZoneId): Boolean {
-        val expirationDay = card.expirationDay
-        return (expirationDay == null || isOnOrBeforeToday(daysSinceEpochToDate(expirationDay), timezone)) &&
+        return !isExpired(card.expirationDay, timezone) &&
             !card.revoked &&
             isTotpValid(totp, card.totpSecret)
     }
```

Comment:
I feel like we have here a tention between implicitly having static and dynamic cards in a single DB table and still having different means of verifying them.

Reply:
I think having two functions enforces the feeling of tension, but basicly the only difference is that a static card does not have a totp secret 🤷🏼‍♀️